queue/init.lua

```diff
@@ -79,10 +99,15 @@ local function wrap_box_cfg()
         orig_cfg = box.cfg
         box.cfg = cfg_wrapper
     elseif type(box.cfg) == 'table' then
-        -- box.cfg after the first box.cfg call
-        local cfg_mt = getmetatable(box.cfg)
-        orig_call = cfg_mt.__call
-        cfg_mt.__call = cfg_call_wrapper
+        if watchers_supported and box.info.status ~= 'running' then
+            -- Wait for the running state and initialize the queue.
+            fiber.new(running_waiter)
+        else
+            -- box.cfg after the first box.cfg call
+            local cfg_mt = getmetatable(box.cfg)
+            orig_call = cfg_mt.__call
+            cfg_mt.__call = cfg_call_wrapper
+        end
```

Comment:
Maybe we shouldn't wait here the `running` state, but status.is_ro == false instead?
```suggestion
        if box.info.ro_reason == 'config' or not watchers_supported then
            -- box.cfg after the first box.cfg call
            local cfg_mt = getmetatable(box.cfg)
            orig_call = cfg_mt.__call
            cfg_mt.__call = cfg_call_wrapper
        else
            fiber.new(rw_waiter)
        end
```

Reply:
Updated.